### PR TITLE
Document migration CLI sub-command

### PR DESCRIPTION
**Pull request by JohnStrunk**
**Describe what this PR does**
- [x] Document `kubectl-volsync migration`
- [x] Fix references to external migration
- [x] Remove old external sync script

**Is there anything that requires special attention?**

**Related issues:**
Fixes: #154 
Depends on: #141 (docs)

### CODE REVIEW

**Comment by JohnStrunk** [comment]
/hold

Waiting on #141

**Comment by JohnStrunk** [comment]
/unhold

#141 has merged, so this is good to go.

**Comment by JohnStrunk** [comment]
/retest

Still can't get a working cluster...

**Comment by JohnStrunk** [comment]
/cc @v-harihar 
/assign @v-harihar 

With the exception of removing the old "external sync" script, there are only doc changes in this PR. Docs build fine.
Instead of continuing to fight w/ the (optional) e2e-openshift job, could I get a review on this?
Built docs: https://volsync--157.org.readthedocs.build/en/157/

File: docs/usage/cli/migration.rst [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+.. contents:: Example steps
+   :local:
+
+The following example uses the ``kubectl volsync migration`` subcommand to
+migrate data from a stand-alone storage system into a Kubernetes
+PersistentVolumeClaim.
+
+External storage
+  A locally mounted directory tree (could be local disk or network-attached
+  storage such as NFS)
```
Comment (v-harihar) [inline docs/usage/cli/migration.rst:33]: Can we also add GlusterFs ?

Comment (v-harihar) [inline docs/usage/cli/migration.rst:33]:
```suggestion
  storage such as NFS, GlusterFs)
```

Comment (JohnStrunk) [reply docs/usage/cli/migration.rst]: done

**Review by v-harihar — APPROVED** [review]
/lgtm

**Pull request merged**